include/net/lwm2m.h

```diff
@@ -26,6 +27,43 @@
 #define IPSO_OBJECT_TEMP_SENSOR_ID			3303
 #define IPSO_OBJECT_LIGHT_CONTROL_ID			3311
 
+struct lwm2m_ctx;
+struct lwm2m_message;
+
+/* Establish a message timeout callback type */
+typedef void (*lwm2m_message_timeout_cb_t)(struct lwm2m_message *msg);
+
+/**
+ * Internal LwM2M message structure
+ *
+ * @details Structure to track in-flight messages.
+ *
+ * Users should not need to manipulate this structure directly.
+ * This definition is here so that an LwM2M context can have a
+ * pool of messages for the library to use.
+ */
+struct lwm2m_message {
+	/** LwM2M context related to this message */
+	struct lwm2m_ctx *ctx;
+
+	/** ZoAP packet data related to this message */
+	struct zoap_packet zpkt;
+
+	/** Message configuration */
+	const u8_t *token;
+	zoap_reply_t reply_cb;
+	lwm2m_message_timeout_cb_t message_timeout_cb;
+	u16_t mid;
+	u8_t type;
+	u8_t code;
+	u8_t tkl;
+
+	/** ZoAP transmission handling structures */
+	struct zoap_pending *pending;
+	struct zoap_reply *reply;
+	u8_t send_attempts;
```

Comment:
can you reorder a bit the attributes for better alignment?  (the u16_t and u8_t ones. Looks like you can gain 4 bytes I think)